### PR TITLE
ps2: keyboard passthru support

### PR DESCRIPTION
**Pull request by No0ne**
I'm currently implementing this: https://github.com/pikvm/pikvm/issues/476

At the moment only keyboard passthru works.
The GPIOs used are 26=data and 27=clock.

@mdevaev please try it out, if you are happy I'll continue with mouse support.

### CODE REVIEW

**Comment by mdevaev** [comment]
Thanks for that! I don't have a PS/2 keyboard right now, I'll buy one soon and test it.

File: hid/pico/src/ph_ps2.c [hunk 1]
```diff
@@ -123,13 +126,13 @@ void ph_ps2_mouse_send_wheel(s8 h, s8 v) {
 
 void ph_ps2_send_clear(void) {
 	if (PH_O_IS_KBD_PS2) {
-		for(u8 key = 0xe0; key <= 0xe7; key++) {
-			kb_send_key(key, false, 0);
-		}
+		//for(u8 key = 0xe0; key <= 0xe7; key++) {
```
Comment (mdevaev) [inline hid/pico/src/ph_ps2.c:129]: BTW why you commented this?

Comment (No0ne) [reply hid/pico/src/ph_ps2.c:129]: This doesn't work. The queue is only 8 bytes long, so blasting it with circa 200 bytes of key-up events won't do it. I'll have to think about key tracking so only those keys which where held down get reset with key-up events.

Comment (mdevaev) [reply hid/pico/src/ph_ps2.c:129]: Got it.

**Comment by mdevaev** [comment]
Sup! I finally found PS/2 keyboard and will test it today or tomorrow.

**Comment by mdevaev** [comment]
Sometimes it skips some key presses. Perhaps because CLK sometimes ends in different ones.

RigolDS5
RigolDS6
RigolDS7

**Comment by No0ne** [comment]
Ok I'll continue on that. I did a larger refactor, please make a new **ps2 passthru branch** in the kvmd repo and I'll make a new pull request.
But first I'll get the main branch onto the pre-1.0 release which has major mouse support fixes.

**Comment by mdevaev** [comment]
:ok_hand:

**Pull request closed without merging**